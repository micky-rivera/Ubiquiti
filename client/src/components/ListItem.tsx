import React from 'react';

function ListItem({name,line, deviceId}: ListItemProps) {

  const resolution = '25x25';

  const imageUrl = `https://static.ui.com/fingerprint/ui/icons/${deviceId}_${resolution}.png`;

  return (
    <>
      <div className='list__item--img'>
        <img src={imageUrl} />
      </div>
      <p className='list__item--line'>{line}</p>
      <p className='list__item--name'>{name}</p>
    </>
  );
}

export default ListItem;
