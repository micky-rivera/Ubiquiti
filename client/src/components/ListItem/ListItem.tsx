import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from '../../hooks/hooks';
import { setChosenProduct } from '../../slices/slices';

function ListItem({name,line, deviceId, details}: ListItemProps) {
  const state = useAppSelector(state => state.app);
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const resolution = state.format === 'list' ? '25x25' : '129x129';

  const imageUrl = `https://static.ui.com/fingerprint/ui/icons/${deviceId}_${resolution}.png`;

  const thisProduct = {
    name: name,
    line: line,
    deviceId: deviceId,
    details: details,
  }

  const handleClick = () => {
    dispatch(setChosenProduct(thisProduct));
    navigate('/product');
  }

  return (
    <>
      <div onClick={handleClick} className={state.format === 'list' ? 'list-item' : 'grid-item'} data-testid='list-item'>
        <div className={state.format === 'list' ? 'list-item__img' : 'grid-item__img'}>
          <img src={imageUrl} data-testid='img' />
        </div>
        <p className={state.format === 'list' ? 'list-item__line' : 'hidden'}>{line}</p>
        <p className={state.format === 'list' ? 'list-item__name' : 'hidden'}>{name}</p>
        <p className={state.format === 'grid' ? 'grid-item__name' : 'hidden'}>{name}</p>
        <p className={state.format === 'grid' ? 'grid-item__line' : 'hidden'}>{line}</p>
      </div>
    </>
  );
}

export default ListItem;
