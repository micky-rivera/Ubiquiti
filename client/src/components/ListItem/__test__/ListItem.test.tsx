import React from "react";
import * as ReactDOM from "react-dom";
import ListItem from '../ListItem';
import { Provider } from 'react-redux'
import {store} from '../../../store/store'
import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

afterEach(cleanup);

const name = 'test product';
const line = 'test line';
const deviceId = '06a25b40-ef1f-463a-82d9-13236866ea3d';
const details = [
    {
        label: 'test label',
        content: 'test content'
    }
]

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Provider store={store}><ListItem name={name} line={line} deviceId={deviceId} details={details} /></Provider></BrowserRouter>, div)
})

it('renders the name correctly', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ListItem name={name} line={line} deviceId={deviceId} details={details} /></Provider></BrowserRouter>);
    expect(getByTestId('list-item')).toHaveTextContent(name);
})

it('renders the line correctly', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ListItem name={name} line={line} deviceId={deviceId} details={details} /></Provider></BrowserRouter>);
    expect(getByTestId('list-item')).toHaveTextContent(line);
})

it('renders an image', () => {
    const { getByTestId } = render(<BrowserRouter><Provider store={store}><ListItem name={name} line={line} deviceId={deviceId} details={details} /></Provider></BrowserRouter>);
    expect(getByTestId('img')).toBeInTheDocument();
})

it('matches snapshot', () => {
    const tree = TestRenderer.create(<BrowserRouter><Provider store={store}><ListItem name={name} line={line} deviceId={deviceId} details={details} /></Provider></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})