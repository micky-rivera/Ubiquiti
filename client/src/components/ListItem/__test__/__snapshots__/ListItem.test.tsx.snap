// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="list-item"
  data-testid="list-item"
  onClick={[Function]}
>
  <div
    className="list-item__img"
  >
    <img
      data-testid="img"
      src="https://static.ui.com/fingerprint/ui/icons/06a25b40-ef1f-463a-82d9-13236866ea3d_25x25.png"
    />
  </div>
  <p
    className="list-item__line"
  >
    test line
  </p>
  <p
    className="list-item__name"
  >
    test product
  </p>
  <p
    className="hidden"
  >
    test product
  </p>
  <p
    className="hidden"
  >
    test line
  </p>
</div>
`;
